packages/store/addon/-private/system/references/has-many.js

```diff
@@ -19,19 +21,23 @@ import Reference, { INTERNAL_MODELS } from './reference';
  @extends Reference
  */
 export default class HasManyReference extends Reference {
-  constructor(store, parentInternalModel, hasManyRelationship, key) {
-    super(store, parentInternalModel);
+  constructor(store, parentIMOrIdentifier, hasManyRelationship, key) {
+    super(store, parentIMOrIdentifier);
     this.key = key;
     this.hasManyRelationship = hasManyRelationship;
     this.type = hasManyRelationship.relationshipMeta.type;
-    this.parent = parentInternalModel.recordReference;
-    this.parentInternalModel = parentInternalModel;
+
+    if (RECORD_ARRAY_MANAGER_IDENTIFIERS) {
+      this.parent = internalModelFactoryFor(store).peek(parentIMOrIdentifier).recordReference;
+    } else {
+      this.parent = parentIMOrIdentifier.recordReference;
+    }
 
     // TODO inverse
   }
 
   _resource() {
-    return INTERNAL_MODELS.get(this)?._recordData.getHasMany(this.key);
+    return internalModelForReference(this)?._recordData.getHasMany(this.key);
```

Comment:
Should be able to get recordData via reference

Reply:
Are you suggesting we make a public getter for `recordData` on the Record instance?  

If not, I was assuming here that the only way to get `recordData` was through the InternalModel.  Perhaps there is a better way?